design/labels.go

```diff
@@ -0,0 +1,103 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var label = a.Type("Label", func() {
+	a.Description(`JSONAPI store for the data of a Label. See also http://jsonapi.org/format/#document-resource-object`)
+	a.Attribute("type", d.String, func() {
+		a.Enum("labels")
+	})
+	a.Attribute("id", d.UUID, "ID of label", func() {
+		a.Example("40bbdd3d-8b5d-4fd6-ac90-7236b669af04")
+	})
+	a.Attribute("attributes", labelAttributes)
+	a.Attribute("relationships", labelRelationships)
+	a.Attribute("links", genericLinks)
+	a.Required("type", "attributes")
+})
+
+var labelAttributes = a.Type("LabelAttributes", func() {
+	a.Description(`JSONAPI store for all the "attributes" of a Label. See also http://jsonapi.org/format/#document-resource-object-attributes`)
+	a.Attribute("name", d.String, "The Label name", nameValidationFunction)
+	a.Attribute("created-at", d.DateTime, "When the label was created", func() {
+		a.Example("2016-11-29T23:18:14Z")
+	})
+	a.Attribute("updated-at", d.DateTime, "When the label was updated", func() {
+		a.Example("2016-11-29T23:18:14Z")
+	})
+	a.Attribute("version", d.Integer, "Version for optimistic concurrency control (optional during creating)", func() {
+		a.Example(23)
+	})
+	a.Attribute("color", d.String, "Color in hex code format. See also http://www.color-hex.com", func() {
+		a.Example("#ffa7cb")
+	})
+})
+
+var labelRelationships = a.Type("LabelRelations", func() {
+	a.Attribute("space", relationGeneric, "This defines the owning space")
+})
+
+var labelList = JSONList(
+	"Label", "Holds the list of Labels",
+	label,
+	pagingLinks,
+	meta)
+
+var labelSingle = JSONSingle(
+	"Label", "Holds a single Label",
+	label,
+	nil)
+
+var _ = a.Resource("label", func() {
+	a.Parent("space")
+	a.BasePath("/labels")
+
+	a.Action("show", func() {
+		a.Routing(
+			a.GET("/:id"),
+		)
+		a.Description("Retrieve label for the given id.")
+		a.Params(func() {
+			a.Param("id", d.String, "id")
+		})
+		//a.UseTrait("conditional")
```

Comment:
Hi @baijum were you able to solve error related to this?
If not, then can you share error details here and maybe @kwk can give some light on that.
Just wanted to track it here and I would like to know the issue. Thanks in advance.

Reply:
@pranavgore09 @baijum sorry for being late. What was the issue here? I guess it has something to do with modifying the `gosupport/conditional_request/generator.go`.

Make sure to add the labels here accordingly:

```go
packageAliases = map[string]string{
		"workitemdsl":     "github.com/fabric8-services/fabric8-wit/workitem",
		"workitemlinkdsl": "github.com/fabric8-services/fabric8-wit/workitem/link",
		"spacedsl":        "github.com/fabric8-services/fabric8-wit/space",
		"iterationdsl":    "github.com/fabric8-services/fabric8-wit/iteration",
		"accountdsl":      "github.com/fabric8-services/fabric8-wit/account",
		"areadsl":         "github.com/fabric8-services/fabric8-wit/area",
		"commentdsl":      "github.com/fabric8-services/fabric8-wit/comment",
	}
	// model structures and their corresponding package alias
	structPackages = map[string]string{
		"WorkItem":         "workitemdsl",
		"WorkItemType":     "workitemdsl",
		"WorkItemLink":     "workitemlinkdsl",
		"WorkItemLinkType": "workitemlinkdsl",
		"Space":            "spacedsl",
		"Iteration":        "iterationdsl",
		"User":             "accountdsl",
		"Identity":         "accountdsl",
		"Area":             "areadsl",
		"Comment":          "commentdsl",
	}
```

Also, make sure that your repository for label is not just `label.Repository` but `label.LabelRepository`. Otherwise the struct above burbs ;) @xcoulon correct?

Reply:
Ouch this one! thanks, @kwk , I will check in my current work and revert!
label.LabelRepository : linter does not like it and recommends to name it label.Repository :laughing:

Reply:
yes, indeed, you'll need to had support for `Label` in the generator, too.